src/orion/executor/multiprocess_backend.py

```diff
@@ -167,13 +167,18 @@ def __init__(self, n_workers=-1, backend="multiprocess", **kwargs):
         if n_workers <= 0:
             n_workers = multiprocessing.cpu_count()
 
+        self.pool_config = {"n_workers": n_workers, "backend": backend}
         self.pool = PoolExecutor.BACKENDS.get(backend, ThreadPool)(n_workers)
 
     def __setstate__(self, state):
-        self.pool = state["pool"]
+        log.warning("Nesting multiprocess executor")
+        self.pool_config = state["pool_config"]
+        backend = self.pool_config.get("backend")
+        n_workers = self.pool_config.get("n_workers", -1)
+        self.pool = PoolExecutor.BACKENDS.get(backend, ThreadPool)(n_workers)
```

Comment:
I'm a bit unsure about this part. If the object is serialized and passed to the subprocess, the deserialization step will have the effect or creating another pool of n_workers, no?

Reply:
yea

Reply:
we maybe able to pass a queue instead to avoid creating multiple pools
but nesting the executor in general is a bit of a nono